src/components/MAPI/clusters/Clusters.tsx

```diff
@@ -89,62 +60,48 @@ const Clusters: React.FC<{}> = () => {
     : null;
 
   const {
-    data: clusterList,
+    data: clusterListData,
     error: clusterListError,
     isValidating: clusterListIsValidating,
-  } = useSWR<ClusterList, GenericResponseError>(clusterListKey, () =>
-    fetchClusterList(clientFactory, auth, provider, namespace, selectedOrgID)
-  );
-
-  useEffect(() => {
-    if (clusterListError) {
-      ErrorReporter.getInstance().notify(clusterListError);
-    }
-  }, [clusterListError]);
-
-  const providerClusterKey = clusterList
-    ? fetchProviderClustersForClustersKey(clusterList.items)
-    : null;
-
-  const {
-    data: providerClusterList,
-    error: providerClusterListError,
-    isValidating: providerClusterListIsValidating,
-  } = useSWR<IProviderClusterForClusterName[], GenericResponseError>(
-    providerClusterKey,
+  } = useSWR<IClusterListWithProviderClusters, GenericResponseError>(
+    clusterListKey,
     () =>
-      fetchProviderClustersForClusters(clientFactory, auth, clusterList!.items)
+      fetchClusterListWithProviderClusters(
```

Comment:
We're storing the response of `fetchClusterListWithProviderClusters()` with the key returned by `fetchClusterListKey()`. I think this could create problems when we try to access the cached data in other parts of the app, because the responses of `fetchClusterListWithProviderClusters()` and `fetchClusterList()` have different interfaces. Could we create a different key for this new fetch function?

Reply:
good find, will fix

Reply:
I decided to go back to two separate `useSWR` calls to keep consistency with other places. I had to change the way we map results of two calls - I added `clustersRef` ref where I store the previous value and return it when the first fetch has finished and the second one is still in progress.